ckan/lib/uploader.py

```diff
@@ -192,6 +194,25 @@ def upload(self, max_size=2):
             except OSError:
                 pass
 
+    def verify_type(self):
+        if not self.filename:
```

Comment:
I suggest skipping all this code entirely if the `ckan.upload.{self.object_type}.types` config options are not set.